packages/legend-extension-dsl-data-space/src/graph-manager/protocol/pure/v1/engine/analytics/V1_DataSpaceAnalysis.ts

```diff
@@ -263,6 +266,8 @@ export class V1_DataSpaceDiagramAnalysisResult {
 }
 
 export abstract class V1_DataSpaceExecutableInfo {
+  id?: string;
```

Comment:
set it to be optional to make sure it's backward compatible as executable that uses service pointer might not have an ID or executionContextKey